Automattic/jetpack
# Update/replace stats get from api usages with stats package methods

#### Changes proposed in this Pull Request:

* Replace usages of stats_get_from_restapi with corresponding methods from Stats package, `WPCOM_Stats `
After discussion in #26708 the methods still return an array. The conversion to `stdclass` object has been implemented by encoding and decoding. To avoid the issue with cached results which may come from both places, the cache prefix has been changed. 
* Fixed  `Automattic\Jetpack\Stats\WPCOM_Stats->fetch_remote_stats`  so that args are coming as a query string in the url.
Filtering of arg keys down the road when using `Automattic\Jetpack\Connection\Client::wpcom_json_api_request_as_blog` was making args not being considered.

* Added get_visits method to the `Automattic\Jetpack\Stats\WPCOM_Stats` class

#### Other information:


#### Jetpack product discussion


1203070774179820-as-1203101231018817/f

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






There are 4 places where the usages of stats_get_from_restapi have been changed to the methods in stats_package.
Create a JN and test the different scenarios with the PR's branch and with trunk. Nothing should change.

1. `get_stats_data` in `class.jetpack-core-api-module-endpoints.php`: This gets triggered when navigating to the Jetpack dashboard in wp-admin. (`/wp-admin/admin.php?page=jetpack#/dashboard`) Make sure stats for Day, Month and week are working fine.
![image](https://user-images.githubusercontent.com/13239096/195426540-63e2f982-c7b8-4044-96ef-4e538fe6bb48.png)

2. `get_benefits` in `class.jetpack-core-api-site-endpoints.php`. This gets triggered when navigating to the Jetpack dashboard in wp-admin as well. (`/wp-admin/admin.php?page=jetpack#/dashboard`) But have not been able to see where the information is being used in the UI. 

The two remaining tests are related to widgets so prior to testing, it is needed to:
- In in Jetpack settings -> Writing; Activate Make extra widgets available for use on your site including subscription forms and Twitter streams 
- Go to Appearance > Widgets and add the corresponding widget ("Top Posts & Pages" Widget and "Blog Stats" Widget)

3. `get_stats` in `widgets/blog-stats.php`. Navigate to a post and check that the widget information is displaying the information.
4. `get_by_views` in `widgets/top-posts.php`. Navigate to a post and check that the widget information is displaying the information.
